wpi.sh

```diff
@@ -0,0 +1,173 @@
+#!/bin/bash
+
+# This script performs WPI via dljc on a given project directory.
+# The inputs are similar to wpi-many.sh, which uses this script internally.
+# The only difference is that wpi-many.sh takes a list of projects, while
+# this script operates on a single project at a time.
+# See the documentation of wpi-many.sh for information on the inputs to this
+# script.
+#
+# Input differences compared to wpi-many.sh:
+# -i and -o are not valid options
+# new required option -d: the directory containing the target project
+#
+# The DLJC environment variable must point to the dljc script.
+
+while getopts "d:u:t:" opt; do
+  case $opt in
+    d) DIR="$OPTARG"
+       ;;
+    u) USER="$OPTARG"
+       ;;
+    t) TIMEOUT="$OPTARG"
+       ;;        
+    \?) # echo "Invalid option -$OPTARG" >&2
+       ;;
+  esac
+done
+
+# shift so that the other arguments (that should be passed to dljc) are all
+# that's in $@
+shift $(( OPTIND - 1 ))
+
+# check required arguments and environment variables:
+
+# testing for JAVA8_HOME, not an unintentional reference to JAVA_HOME
+# shellcheck disable=SC2153
+if [ "x${JAVA8_HOME}" = "x" ]; then
+    echo "JAVA8_HOME must be set to a Java 8 JDK"
+    exit 1
+fi
+
+if [ ! -d "${JAVA8_HOME}" ]; then
+    echo "JAVA8_HOME is set to a non-existent directory ${JAVA8_HOME}"
+    exit 1
+fi
+
+# testing for JAVA11_HOME, not an unintentional reference to JAVA_HOME
+# shellcheck disable=SC2153
+if [ "x${JAVA11_HOME}" = "x" ]; then
+    echo "JAVA11_HOME must be set to a Java 11 JDK"
+    exit 1
+fi
+
+if [ ! -d "${JAVA11_HOME}" ]; then
+    echo "JAVA11_HOME is set to a non-existent directory ${JAVA11_HOME}"
+    exit 1
+fi
+
+JAVA_HOME="${JAVA11_HOME}"
+
+if [ "x${CHECKERFRAMEWORK}" = "x" ]; then
+    echo "CHECKERFRAMEWORK is not set; it must be set to a locally-built Checker Framework. Please clone and build github.com/typetools/checker-framework"
+    exit 2
+fi
+
+if [ ! -d "${CHECKERFRAMEWORK}" ]; then
+    echo "CHECKERFRAMEWORK is set to a non-existent directory ${CHECKERFRAMEWORK}"
+    exit 2
+fi
+
+if [ "x${DLJC}" = "x" ]; then
```

Comment:
Rather than these tests, this script could just clone it (move code from `wpi-many.sh`).